Can't mock a Send Trait

Pseudo cannot mock a trait that is `Send`.  Example:
```rust
    fn send() {
        pub trait A {
            fn foo(&self);
        }

        struct MockA {
            foo: Mock<(), ()>
        }
        impl A for MockA {
            fn foo(&self) {
                self.foo.call(())
            }
        }

        let mock = MockA{foo: Mock::default()};
        let _ = Box::new(mock) as Box<A + Send>;
    }
```
gives the following error:
```
error[E0277]: `(dyn std::ops::Fn(()) + 'static)` cannot be shared between threads safely
   --> src/t_pseudo.rs:431:17
    |
431 |         let _ = Box::new(mock) as Box<A + Send>;
    |                 ^^^^^^^^^^^^^^ `(dyn std::ops::Fn(()) + 'static)` cannot be shared between threads safely                                                   
    |
    = help: the trait `std::marker::Sync` is not implemented for `(dyn std::ops::Fn(()) + 'static)`
    = note: required because of the requirements on the impl of `std::marker::Sync` for `std::ptr::Unique<(dyn std::ops::Fn(()) + 'static)>`
    = note: required because it appears within the type `std::boxed::Box<(dyn std::ops::Fn(()) + 'static)>`
    = note: required because it appears within the type `std::option::Option<std::boxed::Box<(dyn std::ops::Fn(()) + 'static)>>`
    = note: required because of the requirements on the impl of `std::marker::Sync` for `std::sync::RwLock<std::option::Option<std::boxed::Box<(dyn std::ops::Fn(()) + 'static)>>>`
    = note: required because of the requirements on the impl of `std::marker::Send` for `std::sync::Arc<std::sync::RwLock<std::option::Option<std::boxed::Box<(dyn std::ops::Fn(()) + 'static)>>>>`
    = note: required because it appears within the type `pseudo::Mock<(), ()>`
    = note: required because it appears within the type `<t_pseudo::t::Pseudo as TestSuite>::send::MockA`
    = note: required for the cast to the object type `dyn <t_pseudo::t::Pseudo as TestSuite>::send::A + std::marker::Send`
```

--- a/examples/send.rs
+++ b/examples/send.rs
@@ -1,0 +1,25 @@
+// Based on https://github.com/iredelmeier/pseudo/issues/1
+extern crate pseudo;
+
+use pseudo::Mock;
+
+pub trait A {
+    fn foo(&self);
+}
+
+struct MockA {
+    foo: Mock<(), ()>,
+}
+
+impl A for MockA {
+    fn foo(&self) {
+        self.foo.call(())
+    }
+}
+
+fn main() {
+    let mock = MockA {
+        foo: Mock::default(),
+    };
+    let _ = Box::new(mock) as Box<A + Send>;
+}

--- a/src/mock.rs
+++ b/src/mock.rs
@@ -19,7 +19,7 @@ where
 {
     return_value: Arc<RwLock<R>>,
     mock_fn: OptionalRef<fn(C) -> R>,
-    mock_closure: OptionalRef<Box<Fn(C) -> R>>,
+    mock_closure: OptionalRef<Box<Fn(C) -> R + Send + Sync>>,
     calls: Arc<RwLock<Vec<C>>>,
 }
 
@@ -173,7 +173,7 @@ where
     /// assert_eq!(mock.call((1, 1, 1)), 3);
     /// assert_eq!(mock.call((1, 2, 3,)), 6);
     /// ```
-    pub fn use_closure(&self, mock_fn: Box<Fn(C) -> R>) {
+    pub fn use_closure(&self, mock_fn: Box<Fn(C) -> R + Send + Sync>) {
         let mut fn_value = self.mock_fn.write().unwrap();
         *fn_value = None;
 

